### PR TITLE
Issue/56 extract operators from template

### LINKED ISSUE #56: Extract operators from DAG template
Template has grown big recently and entangles quite a bit of logic. Extracting operators would make code more modular and allows reuse of operators.

### PR DESCRIPTION
**Pull request by michalzelechowski-getindata**
Operators are extracted from jinja template to enhance testing capabilities and modularity.

Resolves #56 

---
Keep in mind: 
- [x - in code] Documentation updates
- [x] [Changelog](CHANGELOG.md) updates

### CODE REVIEW

File: kedro_airflow_k8s/operators/node_pod.py [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+        """
+
+        :param node_name: name from the kedro pipeline
+        :param namespace: k8s namespace the pod will execute in
+        :param pvc_name: name of the shared storage attached to this pod
+        :param image: image to be mounted
+        :param image_pull_policy: k8s image pull policy
+        :param env: kedro pipeline configuration name, provided with '-e' option
+        :param task_id: Airflow id to override
+        :param startup_timeout: fter the amount provided in seconds the pod start is
```
Comment (em-pe) [inline kedro_airflow_k8s/operators/node_pod.py:50]: typo `fter` > `after`

**Review by em-pe — COMMENTED** [review]
I like the direction. Minor comments added, plus you'll need to fix `kubernetes` version collision. The rest looks good to me.

File: setup.py [hunk 2]
```diff
@@ -21,6 +21,8 @@
         "tox==3.21.1",
         "pre-commit==2.9.3",
         "apache-airflow==2.0.1",
+        "apache-airflow-providers-cncf-kubernetes==1.1.0",
+        "kubernetes==12.0.1",
```
Comment (em-pe) [inline setup.py:25]: Seems is like a `kubernetes` is a depenedency of `..cncf-kubernetes` - I think we may drop this one.

File: kedro_airflow_k8s/operators/node_pod.py (hunk 1 above)
Comment (michalzelechowski-getindata) [reply kedro_airflow_k8s/operators/node_pod.py]: That's now corrected.

File: kedro_airflow_k8s/operators/create_pipeline_storage.py [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+"""
+from typing import List, Optional
+
+from airflow.operators.python import PythonOperator
+from airflow.utils.decorators import apply_defaults
+from kubernetes import client, config
+from kubernetes.client import models as k8s
+
+
+class CreatePipelineStorageOperator(PythonOperator):
```
Comment (szczeles) [inline kedro_airflow_k8s/operators/create_pipeline_storage.py:13]: There is no need to for the operators to be a subclass of PythonOperator, consider using BaseOperator instead (it's how other Airflow operators do)

**Review by szczeles — APPROVED** [review]
Fantastic work!

File: setup.py (hunk 2 above)
Comment (michalzelechowski-getindata) [reply setup.py]: Well noticed, dropping kubernetes dependency then.

File: kedro_airflow_k8s/operators/create_pipeline_storage.py (hunk 3 above)
Comment (michalzelechowski-getindata) [reply kedro_airflow_k8s/operators/create_pipeline_storage.py:13]: Thanks for the hint, will do that.

**Pull request merged**